File: algorithms/trees/interval-tree/java/IntervalTree.java
public class IntervalTree {
    public static int intervalTree(int[] data) {
        int n = data[0];
        int count = 0;
        int idx = 1;
        for (int i = 0; i < n; i++) {
            int lo = data[idx], hi = data[idx + 1];
            idx += 2;
            int query = data[2 * n + 1];
            if (lo <= query && query <= hi) count++;
        }
        return count;
    }

    public static void main(String[] args) {
        System.out.println(intervalTree(new int[]{3, 1, 5, 3, 8, 6, 10, 4}));
        System.out.println(intervalTree(new int[]{2, 1, 3, 5, 7, 10}));
        System.out.println(intervalTree(new int[]{3, 1, 10, 2, 9, 3, 8, 5}));
    }
}

File: algorithms/trees/interval-tree/python/interval_tree.py
def interval_tree(data):
    n = data[0]
    intervals = []
    idx = 1
    for _ in range(n):
        lo, hi = data[idx], data[idx + 1]
        intervals.append((lo, hi))
        idx += 2
    query = data[idx]

    count = 0
    for lo, hi in intervals:
        if lo <= query <= hi:
            count += 1
    return count


if __name__ == "__main__":
    print(interval_tree([3, 1, 5, 3, 8, 6, 10, 4]))
    print(interval_tree([2, 1, 3, 5, 7, 10]))
    print(interval_tree([3, 1, 10, 2, 9, 3, 8, 5]))
